src/tools/rados/rados.cc

```diff
@@ -3342,18 +3342,25 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
       return 1;
     }
 
-    set<string> out_keys;
-    ret = io_ctx.omap_get_keys(nargs[1], "", LONG_MAX, &out_keys);
-    if (ret < 0) {
-      cerr << "error getting omap key set " << pool_name << "/"
-	   << nargs[1] << ": "  << cpp_strerror(ret) << std::endl;
-      return 1;
-    }
+    string last_read = "";
+    int MAX_READ = 512;
+    do {
+      set<string> out_keys;
+      ret = io_ctx.omap_get_keys(nargs[1], last_read, MAX_READ, &out_keys);
```

Comment:
please use `IoCtx::omap_get_keys2()` instead.

Reply:
ok, have changed it